src/main/java/org/apache/platypus/LuceneServerModule.java

```diff
@@ -0,0 +1,100 @@
+/*
+ *
+ *  *
+ *  *  Copyright 2019 Yelp Inc.
+ *  *
+ *  *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  *  you may not use this file except in compliance with the License.
+ *  *  You may obtain a copy of the License at
+ *  *       http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  *  Unless required by applicable law or agreed to in writing, software
+ *  *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ *  *  either express or implied.
+ *  *  See the License for the specific language governing permissions and
+ *  *  limitations under the License.
+ *  *
+ *  *
+ *
+ *
+ */
+
+package org.apache.platypus;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.AnonymousAWSCredentials;
+import com.amazonaws.auth.profile.ProfileCredentialsProvider;
+import com.amazonaws.auth.profile.ProfilesConfigFile;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.google.inject.AbstractModule;
+import com.google.inject.Inject;
+import com.google.inject.Provides;
+import com.google.inject.Singleton;
+import org.apache.platypus.server.config.LuceneServerConfiguration;
+import org.apache.platypus.server.utils.Archiver;
+import org.apache.platypus.server.utils.ArchiverImpl;
+import org.apache.platypus.server.utils.Tar;
+import org.apache.platypus.server.utils.TarImpl;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static org.apache.platypus.server.config.LuceneServerConfiguration.DEFAULT_BOTO_CFG_PATH;
+
+public class LuceneServerModule extends AbstractModule {
+
+    private final String[] args;
+
+    public LuceneServerModule(String[] args) {
+        this.args = args;
+    }
+
+    @Override
+    public void configure() {
+        bind(Tar.class).to(TarImpl.class);
+    }
+
+    @Inject
+    @Singleton
+    @Provides
+    protected AmazonS3 providesAmazonS3(LuceneServerConfiguration luceneServerConfiguration) {
+        if (luceneServerConfiguration.getBotoCfgPath().equals(DEFAULT_BOTO_CFG_PATH.toString())) {
+            return new AmazonS3Client(new AnonymousAWSCredentials());
+        } else {
+            Path botoCfgPath = Paths.get(luceneServerConfiguration.getBotoCfgPath());
+            final ProfilesConfigFile profilesConfigFile = new ProfilesConfigFile(botoCfgPath.toFile());
+            final AWSCredentialsProvider awsCredentialsProvider = new ProfileCredentialsProvider(profilesConfigFile, "default");
+            return new AmazonS3Client(awsCredentialsProvider);
```

Comment:
This constructor is deprecated, the new way seems to be `AmazonS3ClientBuilder.standard().withCredentials(credentialsProvider).build()`. Not sure if there is a one-liner to use the anonymous credentials though.